src/agent/onefuzz-task/src/tasks/analysis/generic.rs

```diff
@@ -194,11 +194,13 @@ pub async fn run_tool(
     config: &Config,
     reports_dir: &Option<PathBuf>,
 ) -> Result<()> {
+    let target_exe = config.common.setup_dir.join(&config.target_exe);
```

Comment:
To avoid the upgrade breakage described in the PR message, we could temporarily guard this `.join()` on a call to `.starts_with("setup/")`. This does have some edge cases, though, e.g. if a user had a `setup` container which had a nested `target_exe` that (abstractly) resided in `{setup_dir}/setup/fuzz.exe`.

Reply:
I think the guard makes sense -- we go from guaranteed breakage of all inflight 
 WorkUnits  to possible/unlikely breakage. The only thing is, after this change gets deployed, we should remove the guard in the following deploy so we don't keep this edge case around.